libexec/python/docker/tasks.py

```diff
@@ -261,3 +262,48 @@ def get_configs(manifest, keys, delim=None):
     for key in keys:
         configs[key] = get_config(manifest, key, delim=delim)
     return configs
+
+
+def create_bind_tar():
+
+    '''create_bind_tar creates a tar containing empty directories for the
+    user's $HOME and any system-wide "bind path" entries in singularity.conf
+    This allows non-overlay systems to bind these into docker containers
+    when those folders do not already exist in the container.
+    '''
+    tar_file = None
+    files = []
+
+    # Deal with home directory
+    home_bind = os.path.join(os.path.expanduser("~"),
```

Comment:
I think this might also be an issue - if you look at how we get home in defaults.py (https://github.com/singularityware/singularity/blob/master/libexec/python/defaults.py#L107) this was done per a discussion with @gmkurtzer to not use standard `$HOME` for some security reason I can't remember (this was more than a year ago easily!). Regardless of how it's done, this also enforces that using the feature always adds this user home, what if this isn't desired?

Reply:
Hi - thanks for the thoughts on this PR.

I could use that code you referenced or expanduser("~username") here so it never pulls from $HOME, but uses the pwd. Doing "~username" enforces the lookup and ignores $HOME too.

I did actually want to use $HOME for something here.... but it's not generally relevant and should be stripped for consistency.

R.E. always adding the home - I'm trying to keep it simple here - say a user request binding an alternate home at the singularity command line then it'll just be adding a dir here that shouldn't have any impact.